### PR TITLE
migrate to hardhat

### PR DESCRIPTION
**Pull request by ralph-pichler**
makes our framework here more similar to storage-incentives repo

* moves to hardhat (using truffle plugin for test compatibility)
* moves to using signtypeddata_v4 in tests
* moves to yarn instead of npm
* moves to github ci instead of travis

### CODE REVIEW

File: README.md
Comment (Eknir) [inline README.md:24]: I get a bunch of error messages here. Shouldn't we do something about these, especially taking into account upcoming audit
image

File: package.json
Comment (Eknir) [inline package.json:34]: Update this? See line 18 of README

File: README.md
Comment (Eknir) [inline README.md:14]: With node 14, I encounter the following problem:

image


With node 12, everything is ok. I would suggest to mention that we require node 12.

File: package.json
Comment (Eknir) [inline package.json:30]: I get the following warnings:
image

**Review by Eknir — APPROVED** [review]
I verified that everything works as expected -- it does. There are some minor issues which may be tackled now, or otherwise considered for a separate issue. I would like @vandot s feedback on `.github/workflows` before merging.

File: README.md
Comment (ralph-pichler) [reply README.md:24]: let's address any contract changes in the proxy pr. this one is only about the toolchain.

File: README.md
Comment (ralph-pichler) [reply README.md:14]: fixed now

File: package.json
Comment (ralph-pichler) [reply package.json]: it is updated already.

File: package.json
Comment (ralph-pichler) [reply package.json:29]: license headers don't matter much. but we can add this in another pr if needed. changing the `balance` name would break contract abi which I would want to avoid.

File: .github/workflows/ci.yml
Comment (ralph-pichler) [reply .github/workflows/ci.yml:1]: same ci as in storage-incentives repo.

File: package.json
Comment (Eknir) [reply package.json:29]: I think it would be good to add license headers. Added: https://github.com/ethersphere/swap-swear-and-swindle/issues/122

**Comment by Eknir** [comment]
Thanks for addressing my comments. Let's wait for @vandot to approve the github workflows and then it is good to go!

File: README.md
Comment (Eknir) [inline README.md:18]: use `yarn coverage`

Comment (ralph-pichler) [reply README.md]: fixed. also in ci.

File: .github/workflows/ci.yml
Comment (vandot) [reply .github/workflows/ci.yml:1]: @Eknir syntax and yaml look good, I hope that steps/runs do what is needed

Comment (Eknir) [reply .github/workflows/ci.yml:1]: @ralph-pichler - can this somehow be validated before merging?

Comment (ralph-pichler) [reply .github/workflows/ci.yml:1]: @eknir yes, you can just check the ci output on this PR

**Pull request merged**
